webapp/channels/src/components/new_search/search_box.tsx

```diff
@@ -196,9 +214,28 @@ const SearchBox = forwardRef(
                     }
                 }
             },
-            [providerResults, onClose, selectedOption, onSearch, searchType, searchTerms, updateSearchValue],
+            [providerResults, onClose, selectedOption, onSearch, searchType, searchTeam, searchTerms, updateSearchValue],
         );
 
+        const changeSearchTeam = (selectedTeam: string) => {
```

Comment:
This function doesn't come into play when changing the team from RHS results, which can potentially lead to the `In:<channel>` filter applying to a different team than the one `<channel>` belongs to.

Reply:
Fixed with [d10652c](https://github.com/mattermost/mattermost/pull/29389/commits/d10652ce16fa5e465d73073273d9020933616d89)